src/main/java/uk/gov/hmcts/darts/common/service/impl/DataAnonymisationServiceImpl.java

```diff
@@ -97,8 +122,51 @@ public UserAccountEntity getUserAccount() {
         return userIdentity.getUserAccount();
     }
 
+    void tidyUpTransformedMediaEntities(UserAccountEntity userAccount, CourtCaseEntity courtCase) {
+        courtCase
+            .getHearings()
+            .stream()
+            .map(HearingEntity::getMediaRequests)
+            .flatMap(Collection::stream)
+            .distinct()
+            .peek(mediaRequestEntity -> expiredMediaRequest(userAccount, mediaRequestEntity))
+            .map(MediaRequestEntity::getTransformedMediaEntities)
+            .flatMap(Collection::stream)
+            .distinct()
+            .forEach(this::deleteTransformedMediaEntity)
```

Comment:
this stream is quite long, consider breaking up into logical chunks so that debugging and fault diagnosis is easier.

Reply:
Agreed Updated